File: DynamicKey/AgoraDynamicKey/go/test/DynamicKey_test.go
package test

import (
	"../src/DynamicKey"
     "testing"
)

func Test_DynamicKey(t *testing.T) {
	appID:="970ca35de60c44645bbae8a215061b33"
	appCertificate:="5cfd2fd1755d40ecb72977518be15d3b"
	channelName := "7d72365eb983485397e3e3f9d460bdda"
	unixTs:=uint32(1446455472)
	randomInt:=uint32(58964981)

	key := DynamicKey.Generate(appID, appCertificate, channelName, unixTs, randomInt)
	expected := "870588aad271ff47094eb622617e89d6b5b5a615970ca35de60c44645bbae8a215061b3314464554720383bbf5"
	if key != expected {
         t.Error("Error ")
         t.Error(key)
	}
}

File: DynamicKey/AgoraDynamicKey/cpp/test/DynamicKey_test.cpp
#include "../src/DynamicKey.h"
#include <gtest/gtest.h>
#include <string>

class DynamicKey_test : public testing::Test
{
protected:
    // Initializes the test fixture.
    virtual void SetUp(){}

    // Releases resources allocated by the test fixture.
    virtual void TearDown(){}

public:
    // Tests deterministic DynamicKey generation.
    void test_DynamicKey();
};

// Verifies DynamicKey generation against the expected value.
void DynamicKey_test::test_DynamicKey(){
    auto AppID  = "970ca35de60c44645bbae8a215061b33";
    auto  AppCertificate   = "5cfd2fd1755d40ecb72977518be15d3b";
    auto channelName= "7d72365eb983485397e3e3f9d460bdda";
    auto  unixTs = 1446455472;
    auto  randomInt = 58964981;
    std::string result = agora::tools::DynamicKey::generate(AppID, AppCertificate, channelName, unixTs, randomInt);
    EXPECT_EQ(result, "870588aad271ff47094eb622617e89d6b5b5a615970ca35de60c44645bbae8a215061b3314464554720383bbf5");
}

// Tests deterministic DynamicKey generation.
TEST_F(DynamicKey_test, test_DynamicKey)
{
    test_DynamicKey();
}
